akka-cluster/src/main/scala/akka/cluster/Reachability.scala

```diff
@@ -195,7 +195,7 @@ private[cluster] class Reachability private (
   }
 
   def remove(nodes: Iterable[UniqueAddress]): Reachability = {
-    val nodesSet = nodes.to(immutable.HashSet)
+    val nodesSet: Set[UniqueAddress] = nodes.toSet
```

Comment:
Seems suspicious that it was specifically HashSet if there was no reason for it?

Reply:
hmm, indeed - I thought it was a 'mechanical' change because canBuildFrom was removed, but indeed looking at  d274e039f9c it was explicitly `nodes.to[immutable.HashSet]` already. Perhaps better to keep as is then.

Reply:
It will anyway be a HashSet for more than 4 elements, so I doubt the difference matters at all.
UniqueAddress has a very fast `hashCode` so maybe that could be an advantage, less `equals` calls for small sets.

I wouldn't be worried if we changed this to `toSet`.

Reply:
@patriknw If so can you approve? Thx